management/src/main/java/io/micronaut/management/endpoint/processors/AbstractEndpointRouteBuilder.java

```diff
@@ -49,7 +49,7 @@
 @Internal
 abstract class AbstractEndpointRouteBuilder extends DefaultRouteBuilder implements ExecutableMethodProcessor<Endpoint>, LifeCycle<AbstractEndpointRouteBuilder> {
 
-    private static final Pattern ENDPOINT_ID_PATTERN = Pattern.compile("\\w+");
+    private static final Pattern ENDPOINT_ID_PATTERN = Pattern.compile("([^#?]*)");
```

Comment:
I think this is too wide. 
```suggestion
    private static final Pattern ENDPOINT_ID_PATTERN = Pattern.compile("[\\w-]+");
```

Reply:
Agreed, what pattern do you suggest?

Reply:
Updated code to narrow down the check as per suggestion. Kindly check the latest change.